parlai/core/torch_generator_agent.py

```diff
@@ -680,41 +601,42 @@ def eval_step(self, batch):
         text = [self._v2t(p) for p in preds] if preds is not None else None
         return Output(text, cand_choices)
 
-    def beam_search(
-        self,
-        model,
-        batch,
-        beam_size,
-        start=1,
-        end=2,
-        pad=0,
-        min_length=3,
-        min_n_best=5,
-        max_ts=40,
-        block_ngram=0,
-    ):
+    def _treesearch_factory(self, device):
+        method = self.opt.get('inference', 'greedy')
+        beam_size = self.opt.get('beam_size', 1)
+        if method == 'greedy':
+            return GreedySearch(
+                1,
+                min_length=0,
+                min_n_best=1,
+                padding_token=self.NULL_IDX,
+                bos_token=self.START_IDX,
+                eos_token=self.END_IDX,
+                device=device,
+            )
+        elif method == 'beam':
+            return BeamSearch(
+                beam_size,
+                min_length=self.beam_min_length,
+                min_n_best=self.beam_min_n_best,
+                padding_token=self.NULL_IDX,
+                bos_token=self.START_IDX,
+                eos_token=self.END_IDX,
+                device=device,
+            )
+        else:
+            raise ValueError(f"Can't use inference method {method}")
+
+    def _generate(self, batch, beam_size, max_ts=40):
```

Comment:
curious why here the `max_ts` is 40 whereas earlier in the code you default to `256`

Reply:
This should be updated.